[DCP] Adds utility for converting torch save to dcp

LucasLLC opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #120128
* #120127
* #119816
* __->__ #119815
* #119814
* #119813

as title

Differential Revision: [D53718040](https://our.internmc.facebook.com/intern/diff/D53718040/)

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225

LucasLLC committed:
[DCP] Adds utility for converting torch save to dcp

as title

Differential Revision: [D53718040](https://our.internmc.facebook.com/intern/diff/D53718040/)

[ghstack-poisoned]

fegin commented on torch/distributed/checkpoint/format_utils.py:
```diff
@@ -55,7 +57,6 @@ def set_up_planner(
 def dcp_to_torch_save(
     dcp_checkpoint_dir: Union[str, os.PathLike],
     torch_save_fn: Union[str, os.PathLike],
-    coordinator_rank: int = 0,
```
Let's remove this in the previous PR to avoid the BC linter warning.

fegin commented on torch/distributed/checkpoint/format_utils.py:
```diff
@@ -81,3 +76,21 @@ def dcp_to_torch_save(
+
+
+def torch_save_to_dcp(
+    torch_save_fn: Union[str, os.PathLike], dcp_checkpoint_dir: Union[str, os.PathLike]
```
Same as the previous PR, we can avoid using `fn` to represent function name. We can use `path`.

fegin reviewed (commented):
LGTM, some nits.

LucasLLC committed:
Update on "[DCP] Adds utility for converting torch save to dcp"


as title

Differential Revision: [D53718040](https://our.internmc.facebook.com/intern/diff/D53718040/)

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225

[ghstack-poisoned]

LucasLLC committed:
Update on "[DCP] Adds utility for converting torch save to dcp"


as title

Differential Revision: [D53718040](https://our.internmc.facebook.com/intern/diff/D53718040/)

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225

[ghstack-poisoned]

fegin reviewed (approved):
LGTM, please fix the linter before landing.

LucasLLC committed:
Update on "[DCP] Adds utility for converting torch save to dcp"


as title

Differential Revision: [D53718040](https://our.internmc.facebook.com/intern/diff/D53718040/)

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225

[ghstack-poisoned]

LucasLLC committed:
Update on "[DCP] Adds utility for converting torch save to dcp"


as title

Differential Revision: [D53718040](https://our.internmc.facebook.com/intern/diff/D53718040/)

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225

[ghstack-poisoned]

LucasLLC committed:
Update on "[DCP] Adds utility for converting torch save to dcp"


as title

Differential Revision: [D53718040](https://our.internmc.facebook.com/intern/diff/D53718040/)

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225

[ghstack-poisoned]

LucasLLC commented:
@pytorchbot merge

LucasLLC commented:
@pytorchbot merge

LucasLLC commented:
@pytorchbot merge
